### DISCUSSION TITLE
Version 7.44.0-next.1

### DISCUSSION THREAD

**Discussion by bluebill1049** [react-hook-form/react-hook-form#9951] [General]
📄 Form Component update

- added support for custom `fetcher`

```tsx
// axios
<Form 
  action="/api" 
  method="post"
  fetcher={(action, { values }) => axios(action, values)}
/>

// SWR, Tan Query
<Form 
  action="/api" 
  method="post"
  fetcher={(action, { values }) => mutation(action)}
/>
```

- server error no longer set custom error

Beta documenation: http://localhost:8000/api/useform/form

🗑️ remove JSX message check (#9921)

<hr /><em>This discussion was created from the release <a href='https://github.com/react-hook-form/react-hook-form/releases/tag/v7.44.0-next.1'>Version 7.44.0-next.1</a>.</em>

**Comment by joaopaulomoraes** [comment]
Awesome feature! 🚀

**Reply by bluebill1049** [reply]
Awesome to hear!

**Comment by Whoistolu** [comment]
Great feature 👍🏽

**Reply by bluebill1049** [reply]
Thanks! Give it a try and let us know how it goes. <3

**Comment by mmdcoder1** [comment]
Great job keep going! 🚀

**Reply by bluebill1049** [reply]
Thanks will do 👍

**Comment by agcty** [comment]
When will this land? Would love to use this with Remix

**Reply by bluebill1049** [reply]
you can try it today, I think the API is quite stable at this stage.

https://react-hook-form.com/api/useform/form/
https://github.com/react-hook-form/react-hook-form/discussions/9004

I will let the next version run for some more time to patch bugs or issues

**Comment by NateWritesCode** [comment]
Hello, I was not able to get this working with Remix initially but was able to get it to work by creating the FormData like so:

```
          <Form
            control={control}
            method="post"
            action="/foo"
            fetcher={(action, { event }) => {
              const data = new FormData(event?.target);

              submit(data, {
                action,
                method: "post",
              });
            }}
          >
```
Posting here in case it helps someone

**Reply by bluebill1049** [reply]
thanks for sharing.

**Reply by bluebill1049** [reply]
actually we can supply the formData, i will make it into the next release

**Reply by bluebill1049** [reply]
```tsx
fetcher: (
  action: string,
  payload: {
    values?: TFieldValues;
    method: string;
    event?: React.BaseSyntheticEvent;
    formData: FormData;
    formDataJson: string;
  },
) => Promise<void> | void;
```

both json and formdata will be supplied

**Reply by NateWritesCode** [reply]
That's great. Thanks!
